sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java

```diff
@@ -0,0 +1,161 @@
+package com.microsoft.azure.mobile.push;
+
+import android.content.Context;
+
+import com.microsoft.azure.mobile.MobileCenter;
+import com.microsoft.azure.mobile.channel.Channel;
+import com.microsoft.azure.mobile.ingestion.models.json.LogFactory;
+import com.microsoft.azure.mobile.push.ingestion.models.PushInstallationLog;
+import com.microsoft.azure.mobile.push.ingestion.models.json.PushInstallationLogFactory;
+import com.microsoft.azure.mobile.utils.MobileCenterLog;
+import com.microsoft.azure.mobile.utils.storage.StorageHelper;
+
+import junit.framework.Assert;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import static com.microsoft.azure.mobile.push.Push.PREFERENCE_KEY_PUSH_TOKEN;
+import static com.microsoft.azure.mobile.utils.PrefStorageConstants.KEY_ENABLED;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
+@SuppressWarnings("unused")
+@RunWith(PowerMockRunner.class)
```

Comment:
Ok that's the reason why code coverage fails to report percentage for tests originating from this class: if you use `@RunWith(PowerMockRunner.class)`, it will not cover classes that are in `@PrepareForTest`.

The solution for that is to remove `@RunWith(PowerMockRunner.class)` and instead use:

```
    @Rule
    public PowerMockRule mPowerMockRule = new PowerMockRule();
```

The rule is slower and requires the vm arg `-noverify` to test from studio so that's why we don't use it everywhere, but here you need the rule.